src/python/pants/backend/python/util_rules/partition.py

```diff
@@ -19,15 +22,7 @@
 ResolveName = str
 
 
-class _FieldSetWithResolveAndICs(Protocol):
-    @property
-    def resolve(self) -> PythonResolveField: ...
-
-    @property
-    def interpreter_constraints(self) -> InterpreterConstraintsField: ...
-
-
-_FS = TypeVar("_FS", bound=_FieldSetWithResolveAndICs)
+_FS = TypeVar("_FS", bound=FieldSetWithInterpreterConstraints)
```

Comment:
This reuse of the protocol aligns this code with the setup.py code more closely, which tells me this is a good direction to go in.